src/test/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImplTest.java

```diff
@@ -166,6 +157,27 @@ private void prepareCinvDocumentAndProrate(KualiDecimal totalAmountBilledToDate,
         
         cuContractsGrantsInvoiceDocumentServiceImpl.prorateBill(contractsGrantsInvoiceDocument);
     }
+    
+    private static class TestCuContractsGrantsInvoiceDocumentServiceImpl extends CuContractsGrantsInvoiceDocumentServiceImpl {
```

Comment:
Is this test service and the one below it still needed? If you're using the `Mockito.doXXX()` variants to stub the methods of a spied object, then there's no need to explicitly subclass those objects to make the methods unit-test-friendly.

Reply:
yes, not needed